### DISCUSSION TITLE
Fluent method to add sibling node to parent node

### DISCUSSION THREAD

**Discussion by kinekt4** [downforcetech/fluidxml.php#32] [Questions]
The `switchContext` parameter allows a node to add a sibling node to the current node or a child node to the current node.

Is it possible to use the fluent method to add a sibling node to the parent node?

For example, I would like to produce the following

    <food>
      <menu>
          <pasta>
            <name>Spaghetti</name>
          </pasta>
          <pasta>
            <name>Fusilli</name>
          </pasta>
        </menu>
    </food>

Case where `switchContext` is **false** for Spaghetti node:

    $xml = new FluidXml('food');
    $xml->addChild('menu', true)
        ->addChild('pasta', true)
            ->addChild('name', 'Spaghetti', false)
        ->addChild('pasta', true)
            ->addChild('name', 'Fettuccine');

would produce the following:

    <food>        
      <menu>            
        <pasta>                
          <name>Spaghetti</name>                
          <pasta>                    
            <name>Fettuccine</name>                
          </pasta>            
        </pasta>        
      </menu>    
    </food> 

Case where `switchContext` is **true** for Spaghetti node:

    $xml = new FluidXml('food');
    $xml->addChild('menu', true)
        ->addChild('pasta', true)
            ->addChild('name', 'Spaghetti', true)
        ->addChild('pasta', true)
            ->addChild('name', 'Fettuccine');

would produce the following:

    <food>    
      <menu>        
        <pasta>            
          <name>Spaghetti                 
            <pasta>                    
              <name>Fettuccine</name>                
            </pasta>            
          </name>        
        </pasta>    
      </menu>
    </food>

As you can see, neither produces the correct XML. 

**Question**: Is there a way to do this using add/Child() alone? Or do I need to chain xpath queries prior to adding the nodes?

**Comment by daniele-orlando** [comment]
Hi dear @denormalizer. 

What you want to do can be easily done using relative XPath queries; I've implemented them especially for this reason. `addChild()` alone can't help you in this.

```php
$xml = new FluidXml('food');
$xml->addChild('menu', true)
    ->addChild('pasta', true)
        ->addChild('name', 'Spaghetti')
   ->query('..')
   ->addChild('pasta', true)
       ->addChild('name', 'Fettuccine');
```

which produces exactly what you want

```xml
<food>
  <menu>
      <pasta>
        <name>Spaghetti</name>
      </pasta>
      <pasta>
         <name>Fusilli</name>
       </pasta>
    </menu>
</food>
```

If I can give you a suggestion, when you have a predictable data structure which you want to convert to XML, prefer the array input format.

```php
$xml = new FluidXml('food');

$data = [ 'menu' => [
    [ 'pasta' => [ 'name' => 'spaghetti'  ] ],
    [ 'pasta' => [ 'name' => 'fettuccine' ] ],
]];
    
$xml->addChild($data);
```

which is more readable, easier to generalise and to refactor.

**Comment by kinekt4** [comment]
Hi Daniele,

Thanks for your suggestion. I've been umming and ahhing over which method to go with.

I originally went with the array method then I switched over to the fluid method.

I have a predictable (but configurable) data structure. I also have dynamic data. That means I want to be able to easily replace _spaghetti_ with _bolognese_ 

How would you do it with the array method?

**Comment by daniele-orlando** [comment]
I would generate a dynamic array.
